homeassistant/components/media_player/sonos.py

```diff
@@ -893,7 +893,39 @@ def select_source(self, source):
             if len(fav) == 1:
                 src = fav.pop()
                 self._source_name = src['title']
-                self._player.play_uri(src['uri'], src['meta'], src['title'])
+
+                if 'object.container.playlistContainer' in src['meta']:
+                    self.replace_queue_with_playlist(src)
+                    self._player.play_from_queue(0)
+                else:
+                    self._player.play_uri(src['uri'], src['meta'],
+                                          src['title'])
+
+    @soco_error
+    @soco_coordinator
```

Comment:
Both decorators can be removed. This function is only called from within other functions that are already decorated.